src/main/java/org/spongepowered/common/command/SpongeCommandManager.java

```diff
@@ -299,6 +299,9 @@ public boolean containsMapping(CommandMapping mapping) {
 
     @Override
     public CommandResult process(CommandSource source, String commandLine) {
+        if (!Sponge.getServer().isMainThread()) {
```

Comment:
Wouldn't it make more sense to re-submit the command on the main thread? Much like how packets are handled in `NetHandlerPlayServer`?

Reply:
Granted, it'd be a slight behavioral change, but the behavior change is more of "Hey, instead of crashing we re-submit the command to be processed on the main thread."

Reply:
We could, but given this is non-void, what would we return in this case? Or would we just block the async thread - which admittedly, I'm okay with.

Reply:
Wouldn't it be better to use SpongeImplHooks#isMainThread ?

Reply:
I'd be fine with blocking the async task if anything. Thoughts @Zidane ?

Reply:
Block the async thread.

Reply:
Block the async task